var/spack/repos/builtin/packages/py-scikit-learn/package.py

```diff
@@ -15,11 +15,17 @@ class PyScikitLearn(PythonPackage):
     version('0.20.0', sha256='97d1d971f8ec257011e64b7d655df68081dd3097322690afa1a71a1d755f8c18')
     version('0.19.1', 'b67143988c108862735a96cf2b1e827a')
     version('0.18.1', '6b0ff1eaa5010043895dd63d1e3c60c9')
-    version('0.15.2', 'd9822ad0238e17b382a3c756ea94fe0d')
-    version('0.16.1', '363ddda501e3b6b61726aa40b8dbdb7e')
     version('0.17.1', 'a2f8b877e6d99b1ed737144f5a478dfc')
-    version('0.13.1', 'acba398e1d46274b8470f40d0926e6a4')
+    version('0.16.1', '363ddda501e3b6b61726aa40b8dbdb7e')
+    version('0.15.2', 'd9822ad0238e17b382a3c756ea94fe0d')
+
+    depends_on('python@2.7:2.8,3.4:3.7.9999', when='@0.20.0')
+    depends_on('python@2.7:2.8,3.3:3.6.9999', when='@0.19.1')
+    depends_on('py-numpy@1.8.2:', when='@0.19.1:20.0', type=('build', 'run'))
```

Comment:
I usually put 1.8.2:1.99 for numpy, not that is uses semantic versioning, but it at least tries to not break things between majors:
https://www.numpy.org/neps/nep-0023-backwards-compatibility.html

Reply:
SciPy 1.1.0 is out available as a Spack package and scikit-learn seems to work with it.  (I ran a few example scripts.)  It looks like the SciPy folks are trying to maintain backwards compatibility within major versions, so I'll treat it like numpy and set an upper bound of 1.99.